src/Workspaces/Core/Portable/Classification/ClassifierHelper.cs

```diff
@@ -157,6 +171,37 @@ private static void AdjustSpans(ArrayBuilder<ClassifiedSpan> spans, TextSpan wid
             }
         }
 
+        private static void AdjustSpans(ImmutableSegmentedList<ClassifiedSpan>.Builder spans, TextSpan widenedSpan)
```

Comment:
📝 The overloads for `ArrayBuilder<T>` could be removed if we decide to update `AddSyntacticClassificationsAsync` as well.

Reply:
was this a move?

Reply:
Duplicated from the method above it

Reply:
oh, we only did this for semantic classification?  that seems very odd as semnatic classification generally has *much* less classification than syntactic.  (syntactic classifies all tokens, semantic only classifies the semantic side of things).  I would far prefer we keep this all in sync.

Reply:
So far the pull request is a targeted fix for the feedback ticket. Syntactic classification isn't showing up as problematic. Only semantic classification is this expensive.

Reply:
This adds both maintenance burden to us (as now anything working with classification has to potentially handle these two paths), and an engineering burden (helpers need to be duplicated).  It should not be hard to just move this all forward, and it also mitigates any future area where some caller is problematically syntactically classifying too much.

I would be ok with this change if it didn't involve this sort of duplication.  But this is just not palatable.

Reply:
note: i'm also ok and making this change myself in your branch.  but it really is important to me for when this merges to have consistent patterns/behavior across all these methods.

Reply:
This work was intentionally deferred (as mentioned in https://github.com/dotnet/roslyn/pull/67098#discussion_r1120616853) so we can focus the initial discussion on the measured performance problem.

At this time, it's not clear this change is needed, as the source of the problem may be in usage of the API and not with the implementation.